packages/typescript-estree/src/node-utils.ts

```diff
@@ -602,6 +605,16 @@ export function convertToken(
       value,
     };
   }
+
+  if (tokenType === AST_TOKEN_TYPES.PrivateIdentifier) {
+    return {
+      type: tokenType,
+      loc,
+      range,
+      value: value.slice(1),
```

Comment:
I've tried to reproduce the same behavior in: https://github.com/typescript-eslint/typescript-eslint/blob/f0072dc5f00fc24815fc25ad0e645412c1b84f38/packages/typescript-estree/src/convert.ts#L968